tests/foreman/api/test_repository.py

```diff
@@ -279,18 +286,23 @@ def test_create_docker_repo(self, name):
         @Feature: Repository
 
         """
-        product = entities.Product(organization=self.org).create()
-        repo = entities.Repository(
-            content_type=u'docker',
-            docker_upstream_name=u'busybox',
-            name=name,
-            product=product,
-            url=DOCKER_REGISTRY_HUB,
-        ).create()
-        repo2 = repo.read()
-        self.assertEqual(repo.name, repo2.name)
-        self.assertEqual(repo.docker_upstream_name, repo2.docker_upstream_name)
-        self.assertEqual(repo.content_type, repo2.content_type)
+        for name in _valid_names():
+            with self.subTest(name):
+                product = entities.Product(organization=self.org).create()
```

Comment:
We could move this outside the loop and have all repositories re-use the one product.